listenbrainz/db/feedback.py

```diff
@@ -0,0 +1,152 @@
+import sqlalchemy
+
+from listenbrainz import db
+
+
+def insert(user_id, recording_msid, score):
+    """ Inserts a feedback record for a user's loved/hated recording into the database.
+        If the record is already present for the user, the score is updated to the new
+        value passed.
+        Args:
+            user_id (int): the row id of the user
+            recording_msid (string): the MessyBrainz ID of the recording
+            score (int): The score associated with the recording (+1/-1 for love/hate respectively)
+    """
+
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            INSERT INTO recording_feedback (user_id, recording_msid, score)
+                 VALUES (:user_id, :recording_msid, :score)
+            ON CONFLICT (user_id, recording_msid)
```

Comment:
Did we want to consider if a user's feedback for a song changes over time? Maybe they used to like it, but no longer do. Or they like it in the mornings, but not in the evenings, or they only want it when they're in a particular mood.
These are all possible future uses of this data, and I understand if we don't want to support it right now, but perhaps it could be interesting to store multiple feedbacks rather than overwriting them?

Reply:
I am not quite sure about allowing multiple feedbacks. I mean if we want to know how the feedbacks for a recording drift over time, we may like to allow multiple records. 
I would like to know what @mayhem and @paramsingh have to say about this?

Reply:
We could store multiple rows and just show the latest one on the listen page. I think storing as much of this data as possible might be a good idea?

Reply:
As per IRC discussion, I think this data is going to be sparse and we shouldn't bother collecting it now. If we need it later, then we can always turn it on. Let's keep the on conflict clause.